core/src/test/java/org/springframework/security/authentication/DefaultAuthenticationEventPublisherTests.java

```diff
@@ -137,6 +137,28 @@ public void unknownFailureExceptionIsIgnored() {
 		verifyZeroInteractions(appPublisher);
 	}
 
+	@Test(expected = IllegalArgumentException.class)
+	public void defaultAuthenticationFailureEventClassSetNullThen() {
+		publisher = new DefaultAuthenticationEventPublisher();
+		publisher.setDefaultAuthenticationFailureEvent(null);
+	}
+
+	@Test
+	public void defaultAuthenticationFailureEventIsPublished() {
+		publisher = new DefaultAuthenticationEventPublisher();
+		Properties p = new Properties();
+		p.put(MockAuthenticationException.class.getName(),
+				AuthenticationFailureDisabledEvent.class.getName());
+		publisher.setAdditionalExceptionMappings(p);
+		publisher.setDefaultAuthenticationFailureEvent(AuthenticationFailureBadCredentialsEvent.class);
+		ApplicationEventPublisher appPublisher = mock(ApplicationEventPublisher.class);
+
+		publisher.setApplicationEventPublisher(appPublisher);
+		publisher.publishAuthenticationFailure(new AuthenticationException("") {
+		}, mock(Authentication.class));
+		verify(appPublisher).publishEvent(isA(AuthenticationFailureBadCredentialsEvent.class));
+	}
+
 	private static final class MockAuthenticationException extends
```

Comment:
Would you please add a test for a class that is missing the appropriate constructor? That will make sure to test that the exception is thrown appropriately.